lib/cachex/actions/warm.ex

```diff
@@ -26,15 +26,22 @@ defmodule Cachex.Actions.Warm do
   be returned in the result of this call.
   """
   def execute(cache() = cache, options) do
+    respect_async_warmers = Keyword.get(options, :respect_async_warmers, false)
     mods = Keyword.get(options, :modules, nil)
-    parent = Services.locate(cache, Services.Incubator)
-    children = Supervisor.which_children(parent)
 
-    handlers =
-      for {mod, pid, _, _} <- children, mods == nil or mod in mods do
-        send(pid, :cachex_warmer) && mod
-      end
+    case Services.locate(cache, Services.Incubator) do
```

Comment:
Calling `Cachex.warm` on a cache without warmers was throwing errors - this `case` allows these calls to noop instead

Reply:
Yep, another nice catch (I'm amazed the tests didn't show this...). Rather than switching all of this into a `case` though, a more elegant check is:

```elixir
children = if parent, do: Supervisor.which_children(parent), else: []
```

With this the only diff in this function becomes the `respect_async_warmers` lines, which is nice.